roachtest: fix ycsb regression by giving workload node more cpu

DarrylWong opened the pull request:
Previously, we gave the workload node only 4 CPUs but this
showed perf regressions and ocassional flakes on workload E
which is read heavy.

This change reverts that and gives ycsb workload nodes the
full CPUs as the rest of the cluster. Some of the more write
heavy tests seemed fine with just 4 CPUs, but for consistency
we revert it for all workloads.

Fixes: #128145
Release note: none
Epic: none

DarrylWong committed:
roachtest: fix ycsb regression by giving workload node more cpu

Previously, we gave the workload node only 4 CPUs but this
showed perf regressions and ocassional flakes on workload E
which is read heavy.

This change reverts that and gives ycsb workload nodes the
full CPUs as the rest of the cluster. Some of the more write
heavy tests seemed fine with just 4 CPUs, but for consistency
we revert it for all workloads.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/128334)
<!-- Reviewable:end -->

DarrylWong committed:
workload: log workload errors

This change logs workload errors in addition to
returning them if tolerate errors is false. This
displays a stack trace whic just returning does not.
